Gemfile

```diff
@@ -3,3 +3,4 @@ source "https://rubygems.org"
 gemspec
 
 gem 'puppet', ENV['PUPPET_VERSION'] || '>=2.7.16'
+gem 'public_suffix', '< 3' if RUBY_VERSION < '2.1'
```

Comment:
Can you help me find where this is being used in the newly added code?

Reply:
I believe it's a dependency of a dependency. Sadly bundler is unable to look at supported Ruby versions when doing dependency resolution you so end up with these horrible hacks. One of the big things I hate about the Ruby ecosystem.